qa/acceptance/spec/lib/cli/logstash-plugin/install_spec.rb

```diff
@@ -0,0 +1,74 @@
+# encoding: utf-8
+require_relative "../../../spec_helper"
+require "logstash/version"
+require "fileutils"
+require "logstash/devutils/rake"
+
+shared_examples "logstash install" do |logstash|
+  before(:each) do
+    logstash.install(LOGSTASH_VERSION)
+  end
+
+  after :each do
+    logstash.uninstall
+  end
+
+  describe "on #{name}" do
+    context "with a direct internet connection" do
+      context "when the plugin exist" do
+        context "from a local `.GEM` file" do
+          let(:gem_path) { download_gem("logstash-filter-dns", "2.1.3") }
+          let(:gem_path_on_vagrant) { "/vagrant/vendor/download/logstash-filter-dns-2.1.3.gem" }
+          # after(:each) { FileUtils.rm_rf(gem_path) }
+
+          it "successfully install the plugin" do
+            command = logstash.run_command_in_path("bin/logstash-plugin install #{gem_path_on_vagrant}")
+            expect(command).to run_successfully_and_output(/Installation successful/)
+            expect(logstash.plugin_installed?(logstash.host, "logstash-filter-dns")).to be_truthy
+          end
+        end
+
+        context "when fetching a gem from rubygems" do
+          it "successfully install the plugin" do
+            command = logstash.run_command_in_path("bin/logstash-plugin install logstash-filter-dns")
+            expect(command).to run_successfully_and_output(/Installation successful/)
+            expect(logstash.plugin_installed?(logstash.host, "logstash-filter-dns")).to be_truthy
+          end
+
+          it "allow to install a specific version" do
+            command = logstash.run_command_in_path("bin/logstash-plugin install --version 2.1.2 logstash-filter-dns")
+            expect(command).to run_successfully_and_output(/Installation successful/)
+            expect(logstash.plugin_installed?(logstash.host, "logstash-filter-dns", "2.1.2")).to be_truthy
+          end
+        end
+      end
+
+      context "when the plugin doesnt exist" do
+        it "fails to install and report an error" do
+          # LOOK WITH PERE WHY EXITSTATUS IS NIL
+          command = logstash.run_command_in_path("bin/logstash-plugin install --no-verify logstash-output-impossible-plugin")
+          # expect(command).to fail_and_output(/Plugin not found, aborting/)
+          expect(command.stderr).to match(/Plugin not found, aborting/)
+        end
+      end
+    end
+  end
+end
+
+describe "CLI/logstash-plugin install" do
+  config = ServiceTester.configuration
+  config.servers.each do |address|
+    logstash = ServiceTester::Artifact.new(address, config.lookup[address])
+    it_behaves_like "logstash install", logstash
+  end
+end
+
+DOWNLOAD_DIRECTORY = File.join(File.dirname(__FILE__), "..", "..", "vendor", "download").freeze
+def download_gem(name, version)
```

Comment:
might be something to wrap inside of a helpers file? so test are more just the specs definition.